git_wrapper/repo.py

```diff
@@ -189,3 +189,90 @@ def commit(self, new_commit):
         if not isinstance(new_commit, GitCommit):
             raise TypeError("Commit must be a GitCommit object.")
         self._commit = new_commit
+
+    def sync(self, remote="origin"):
+        """Refresh (fetch) the specified remote
+
+           :param str remote: Remote to sync
+        """
+        try:
+            remote = self.repo.remote(remote)
+        except ValueError:
+            msg = "Remote {remote} does not exist on repo {repo}".format(
+                remote=remote,
+                repo=self.repo.working_dir
+            )
+            raise exceptions.ReferenceNotFoundException(msg)
+
+        try:
+            remote.fetch()
+        except git.GitCommandError as ex:
+            msg = (
+                "Could not fetch remote {remote} ({url}). "
+                "Error: {error}".format(remote=remote.name,
+                                        url=remote.url,
+                                        error=ex)
+            )
+            raise_from(exceptions.RemoteException(msg), ex)
+
+    def sync_all(self):
+        """Refresh (fetch) all the repo's remotes.
+
+           All the remotes will be fetched even if one fails ; in this case a
+           single exception containing the list of failed remotes is returned.
+        """
+        remotes = self.remote.names()
+
+        errors = []
+        for remote in remotes:
+            try:
+                self.sync(remote)
+            except exceptions.RemoteException:
+                self.logger.exception("Error fetching remote %s", remote)
+                errors.append(remote)
+
+        if errors:
+            msg = "Error fetching these remotes: {0}".format(", ".join(errors))
+            raise exceptions.RemoteException(msg)
+
+    def reset(self, branch="master", remote="origin", remote_branch="master",
```

Comment:
On thought here is to allow for a 'type' parameter the defaults to hard. We can then extend this to allow for a soft reset if we have a need later. Otherwise I would recommend making it clear this is a hard reset by changing the name.

Reply:
Good point! I'll rename.

I'm also realising that although I was thinking about this in terms of "refreshing a repo", the sync() methods actually would fit better under "remote", while reset() applies to a "branch" so I'll move things around a bit in the next patch set.